unixfs/archive/archive.go

```diff
@@ -53,6 +66,11 @@ func DagArchive(ctx cxt.Context, nd *mdag.Node, name string, dag mdag.DAGService
 				pipew.CloseWithError(err)
 				return
 			}
+			maybeGzw.Close()
```

Comment:
explicit Close is needed for a gzip writer. Reading the source, it write the end record etc.